app/mailers/devise_notify_mailer.rb

```diff
@@ -41,6 +45,11 @@ def confirmation_instructions(record, token, _opts = {})
     )
   end
 
+  def intercept_send?(email)
+    email_domain = email.split("@").last.downcase
+    !(Rails.env.production? || Rails.env.test?) && !ALLOWLIST.include?(email_domain)
```

Comment:
I inverted logic to Rails.env.staging? && ALLOWLIST.exclude?(email_domain) and got tests passing and probably easier to read as well

Reply:
My reasoning for this way round was that if another environment was added (sandbox for example) you probably want that to default to the block behaviour rather than the allow behaviour?

Will change the include to exclude though.

Reply:
That I did not see and it makes sense!